### ISSUE TITLE
Adding a custom social media item with logo

### ISSUE THREAD

**Issue by respatte** [xdanaux/moderncv#74]
Hi,

Not really an issue, but I was wondering if it would be possible to create a custom `\social` item where the user can provide his own logo for a social media. I'm guessing it would be possible to tweak the source files to create one, but I'm not big on using TeX like that...

The reason I'm asking that is that I would like to give my ORCID ID with their nice little logo.

If creating a custom command is too much work for the use, I would love it if you could guide me through the steps of doing it myself!

Best,
Arthur

**Comment by mcala** [comment]
If you still want to give this a go yourself, it's not too bad. However, ORCID's logo isn't included in [FontAwesome](https://fontawesome.com/icons?d=gallery&q=ORCID), which is how the other icons are typeset. So the *more difficult* part is figuring out how to get the icon to show up in a nice way. I'd probably just stick with writing out ORC ID. If you want to still give it a shot:

First you'll want to make your own copy of the moderncv project. On a Mac with MacTeX, you'd put this in: `/Users/${YOUR_USERNAME}/Library/texmf/tex/latex/moderncv`

This will keep your work for being overwritten if the package updates from your package manager. You can also keep the files in the same directory as your resume/CV if you have trouble finding where to put them.

In the moderncv.cls (class) file is where you'll find what you are changing. Edit this file and find `\collectionnew{socials}`, which is where all of the social icons and links are described. You should see Twitter, Github and some others. Copy one of the `\ifthenelse` lines and modify it to work with ORCID.

Now search for `\newcommand*{\twittersocialsymbol}`. Copy this making something like `\newcommand*{\orcidsocialsymbol}`. Save and exit this file.

Now you have to tell moderncv what icon to use for ORC ID, which is going to in one of the moderncvicons.sty (style) files. Make sure you edit the one you are using (since you mention the icons, I'd assume it's the FontAwesome one). Like I said, there isn't a FontAwesome icon for ORCID, so you'll probably wind up with something like this: `\renewcommand*{\orcidsocialsymbol} {{\textbf{ORC ID:}~}`

**Comment by khatchad** [comment]
I see the icon now in https://fontawesome.com/icons?d=gallery&q=ORCID. However, when I use `\social[orcid]`, the symbol is still missing.

**Comment by mcala** [comment]
Correct, you still need to go through the rest of my comment to insert a new command for the ORCID symbol. Now that there is a FontAwesome icon, you'd want the last renew command to be: `\renewcommand*{\orcidsocialsymbol} {\small \faorcid}`

**Comment by khatchad** [comment]
Ah, OK. I currently see this in `moderncviconsawesome.sty`:
```
\renewcommand*{\orcidsocialsymbol}        {{\small\aiOrcid}~}
```
